client/src/app/dashboards/dashboard/dashboard.component.ts

```diff
@@ -1,15 +1,46 @@
 import { Component, OnInit } from '@angular/core';
+import { forkJoin } from 'rxjs';
+import {
+  DashboardService,
+  DashboardTotals,
+} from 'src/app/services/dashboard.service';
+import { LogEntry, LoginService } from 'src/app/services/login.service';
 
 @Component({
   selector: 'app-dashboard',
   templateUrl: './dashboard.component.html',
-  styleUrls: ['./dashboard.component.scss']
+  styleUrls: ['./dashboard.component.scss'],
 })
-export class DashboardComponent implements OnInit {
+export class DashboardComponent {
+  logins: LogEntry[] = [];
+  dashboardTotals: any;
+  isLoading: boolean = true;
 
-  constructor() { }
+  constructor(
+    private loginService: LoginService,
+    private dashboardService: DashboardService
+  ) {
+    forkJoin({
+      dashboardTotals: this.dashboardService.getDashboardTotals(),
+      logins: this.loginService.getLogins(),
+    }).subscribe((mergedData) => {
+      this.logins = this.sortByDate(mergedData.logins);
+      this.dashboardTotals = mergedData.dashboardTotals;
+      this.isLoading = false;
+    });
+  }
 
-  ngOnInit(): void {
+  calculatePercentage(dashboardTotals: DashboardTotals) {
+    return Math.round(
+      (this.dashboardTotals.totalFailures /
+        this.dashboardTotals.totalTokensIssued) *
```

Comment:
Really should be divided by a new total "login attempts" or something rather than tokens issued. 

e.g. 
Drop DB, startup and use invalid credentials
Then login as admin, observe the failure rate is 100% rather than 50% (with you now logged in)
Logout and login again, observe the failure rate is 50%. rather than 33%